app/assets/javascripts/discourse/app/controllers/flag.js

```diff
@@ -18,6 +21,55 @@ export default Controller.extend(ModalFunctionality, {
   topicActionByName: null,
   spammerDetails: null,
 
+  flagActions: {
+    icon: "gavel",
+    label: I18n.t("flagging.take_action"),
+    actions: [
+      {
```

Comment:
One thing we did when migrating to the review queue was to have the server side pass back all the actions you could take. This was done to help people integrating with the API because they didn't need to know all the actions on the client side.

Since this is the front end, and a very specific thing (take action) maybe this approach is OK. I just wanted to bring it up in case you had any thoughts about it.

Reply:
Yeah, I initially wanted to query for flagging options, but I didn't see a better option here when I began. My main goal here was to re-use the same widgets we had set up for the review queue.

I went with setting things up through client-side, because there's no reviewable flag to query at the time of flagging. Perhaps one alternative approach would be exposing a way for reviewables to send down a sample of their actions (like this) without having to create an instance of a reviewable?